Build failure: synergy

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

Using the synergy package on nixos-unstable, fails to build:


Must be related to #445447 

### Can Hydra reproduce this build failure?

Hydra doesn’t try to build the package.

### Link to Hydra build job

_No response_

### Relevant log output

```console
@nix { "action": "setPhase", "phase": "configurePhase" }
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_LOCALEDIR=/nix/store/8jmgr5mx6im92j9jsj8w84ql9vk8v3ml-synergy-1.14.6.19-stable/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/8jmgr5mx6im92j9jsj8w84ql9vk8v3ml-synergy-1.14.6.19-stable/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/8jmgr5mx6im92j9jsj8w84ql9vk8v3ml-synergy-1.14.6.19-stable/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/8jmgr5mx6im92j9jsj8w84ql9vk8v3ml-synergy-1.14.6.19-stable/share/doc/synergy-core -DCMAKE_INSTALL_INFODIR=/nix/store/8jmgr5mx6im92j9jsj8w84ql9vk8v3ml-synergy-1.14.6.19-stable/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/8jmgr5mx6im92j9jsj8w84ql9vk8v3ml-synergy-1.14.6.19-stable/share/man -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/8jmgr5mx6im92j9jsj8w84ql9vk8v3ml-synergy-1.14.6.19-stable/include -DCMAKE_INSTALL_SBINDIR=/nix/store/8jmgr5mx6im92j9jsj8w84ql9vk8v3ml-synergy-1.14.6.19-stable/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/8jmgr5mx6im92j9jsj8w84ql9vk8v3ml-synergy-1.14.6.19-stable/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/8jmgr5mx6im92j9jsj8w84ql9vk8v3ml-synergy-1.14.6.19-stable/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/dmypp1h4ldn0vfk3fi6yfyf5yxp9yz0k-gcc-wrapper-14.3.0/bin/strip -DCMAKE_RANLIB=/nix/store/dmypp1h4ldn0vfk3fi6yfyf5yxp9yz0k-gcc-wrapper-14.3.0/bin/ranlib -DCMAKE_AR=/nix/store/dmypp1h4ldn0vfk3fi6yfyf5yxp9yz0k-gcc-wrapper-14.3.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/8jmgr5mx6im92j9jsj8w84ql9vk8v3ml-synergy-1.14.6.19-stable
CMake Error at CMakeLists.txt:17 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.


-- Configuring incomplete, errors occurred!
****
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.49, NixOS, 25.11 (Xantusia), 25.11.git.37f29816e5c3`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.5`
 - channels(root): `"home-manager, nixos"`
 - nixpkgs: `/nix/store/5g1px1rk3l1kichqrlll18d3q5xx5b4p-nixos/nixos`

### Notify maintainers

@talyz 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/misc/synergy/default.nix
+++ b/pkgs/applications/misc/synergy/default.nix
@@ -48,11 +48,14 @@ stdenv.mkDerivation rec {
 
   postPatch = ''
     substituteInPlace src/gui/src/SslCertificate.cpp \
-      --replace 'kUnixOpenSslCommand[] = "openssl";' 'kUnixOpenSslCommand[] = "${openssl}/bin/openssl";'
+      --replace-fail 'kUnixOpenSslCommand[] = "openssl";' 'kUnixOpenSslCommand[] = "${openssl}/bin/openssl";'
+
+    substituteInPlace CMakeLists.txt cmake/Version.cmake src/gui/CMakeLists.txt \
+    --replace-fail "cmake_minimum_required (VERSION 3.4)" "cmake_minimum_required(VERSION 3.10)"
   ''
   + lib.optionalString stdenv.hostPlatform.isLinux ''
     substituteInPlace src/lib/synergy/unix/AppUtilUnix.cpp \
-      --replace "/usr/share/X11/xkb/rules/evdev.xml" "${xkeyboardconfig}/share/X11/xkb/rules/evdev.xml"
+      --replace-fail "/usr/share/X11/xkb/rules/evdev.xml" "${xkeyboardconfig}/share/X11/xkb/rules/evdev.xml"
   '';
 
   nativeBuildInputs = [

